File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/networks/NetworkPaths.java
package dev.reuss.tmdb.domain.networks;

import dev.reuss.tmdb.value.id.NetworkId;

final class NetworkPaths {

    private static final String NETWORK = "/network";

    private NetworkPaths() {
    }

    static String details(NetworkId networkId) {
        return network(networkId);
    }

    static String alternativeNames(NetworkId networkId) {
        return network(networkId) + "/alternative_names";
    }

    static String images(NetworkId networkId) {
        return network(networkId) + "/images";
    }

    private static String network(NetworkId networkId) {
        return NETWORK + "/" + networkId.asString();
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/networks/NetworkPaths.kt
package dev.reuss.tmdb.domain.networks

import dev.reuss.tmdb.core.path.tmdbPath
import dev.reuss.tmdb.value.id.NetworkId

internal object NetworkPaths {

    fun details(networkId: NetworkId): String =
        tmdbPath("network", networkId)

    fun alternativeNames(networkId: NetworkId): String =
        tmdbPath("network", networkId, "alternative_names")

    fun images(networkId: NetworkId): String =
        tmdbPath("network", networkId, "images")
}